backend/src/main/java/com/redhat/cloud/notifications/auth/ConsoleIdentityProvider.java

```diff
@@ -75,19 +80,43 @@ public Class<ConsoleAuthenticationRequest> getRequestType() {
 
     @Override
     public Uni<SecurityIdentity> authenticate(ConsoleAuthenticationRequest rhAuthReq, AuthenticationRequestContext authenticationRequestContext) {
+        // The Kessel back end takes priority over any other authentication, in
+        // order to be able to have both the RBAC and Kessel back ends enabled
+        // at the same time. That way we can jump from using RBAC to Kessel and
+        // if we see that something is not working properly, we can instantly
+        // switch back to RBAC by disabling Kessel.
+        if (this.backendConfig.isKesselBackendEnabled()) {
+            try {
+                // Build the principal from the incoming "x-rh-identity" header.
+                final Optional<Principal> principal = this.buildPrincipalFromIdentityHeader(rhAuthReq);
+
+                final QuarkusSecurityIdentity.Builder builder = QuarkusSecurityIdentity.builder();
+
+                // Set the principal to the one we decoded from the header,
+                // unless no header was present. In that case we build an empty
+                // principal.
+                builder.setPrincipal(principal.orElse(ConsolePrincipal.noIdentity()));
+
+                // Build the security identity for Quarkus.
+                return Uni.createFrom().item(builder.build());
+            } catch (final IllegalIdentityHeaderException e) {
```

Comment:
The `IllegalIdentityHeaderException` is thrown in the `ConsolePrincipalFactory`:

https://github.com/RedHatInsights/notifications-backend/blob/60352b1d5f1b745048070346a039ed38718d0c27/backend/src/main/java/com/redhat/cloud/notifications/auth/principal/ConsolePrincipalFactory.java#L14-L23

If that exception is thrown it will be thrown everywhere, since the "no RBAC" mode also uses the `ConsolePrincipalFactory.fromIdentity` function:

https://github.com/RedHatInsights/notifications-backend/blob/60352b1d5f1b745048070346a039ed38718d0c27/backend/src/main/java/com/redhat/cloud/notifications/auth/ConsoleIdentityProvider.java#L107-L118

And the same thing for when we call RBAC:

https://github.com/RedHatInsights/notifications-backend/blob/60352b1d5f1b745048070346a039ed38718d0c27/backend/src/main/java/com/redhat/cloud/notifications/auth/ConsoleIdentityProvider.java#L137-L149

I wrapped all those calls in this function for convenience, but they all will throw that exception if the header is incorrect:

https://github.com/RedHatInsights/notifications-backend/blob/60352b1d5f1b745048070346a039ed38718d0c27/backend/src/main/java/com/redhat/cloud/notifications/auth/ConsoleIdentityProvider.java#L208-L225